### PR TITLE
[Clang][Sema] Fix Wswitch-default bad warning in template

### PR DESCRIPTION
**Pull request by hstk30-hw**
https://github.com/llvm/llvm-project/pull/73077 added -Wswitch-default diagnostic but it produced false positives in templates. This PR will address that. https://github.com/llvm/llvm-project/issues/75943

### CODE REVIEW

**Comment by shafik** [comment]
Can you add more details to you summary "https://github.com/llvm/llvm-project/pull/73077 added -Wswitch-default diagnostic but it produced false positives in templates. This PR will address that issue"

File: clang/test/Sema/switch-default-template.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -std=c++11 -Wswitch-default %s
```
Comment (shafik) [inline clang/test/Sema/switch-default-template.cpp:1]: I think we can add these test to `clang/test/Sema/switch-default.c`

**Comment by dongjianqiang2** [comment]
thanks for correcting this.

**Review by cor3ntin — APPROVED** [review]
LGTM, thanks

**Review by aaronpuchert — APPROVED** [review]
For now I guess this is Ok, although I think the better fix would be to diagnose missing or duplicate `default` labels even in the dependent case. Because `default` labels themselves are never dependent.

**Comment by shafik** [comment]
> For now I guess this is Ok, although I think the better fix would be to diagnose missing or duplicate `default` labels even in the dependent case. Because `default` labels themselves are never dependent.

We should probably add a FIXME for this.

**Review by dongjianqiang2 — APPROVED** [review]
LGTM, thanks

**Review by aaronpuchert — APPROVED** [review]

**Pull request merged**